cli/azd/internal/repository/infra_confirm.go

```diff
@@ -107,6 +108,11 @@ func (i *Initializer) infraSpecFromDetect(
 			if svc.Language == appdetect.Java {
 				serviceSpec.Port = 8080
 			}
+		} else {
+			ports := i.detectPortInDockerfile(svc.Docker.Path)
```

Comment:
@rujche Let's move this logic into `appdetect` package. The concrete place I'm thinking is insider `docker.go`. We can update `Docker` to include the "ExposedPorts" property as part of the analysis.